### PR TITLE
Add callback signature validation

### PR DESCRIPTION
**Pull request by jtnegrotto**

### CODE REVIEW

File: lib/phaxio/client.rb [hunk 1]
```diff
@@ -271,15 +275,101 @@ def get_account_status
...
+    # options - Type: hash. Information needed to authenticate the callback.
+    #   :url       - Type: string. The full URL that was called by Phaxio,
+    #                including the query. (required)
+    #   :params    - Type: hash. The POSTed form data (required)
+    #   :files     - Type: array. Submitted files (required - "received" fax
+    #                callback only)
+    #   :signature - Type: string. The X-Phaxio-Signature HTTP header value
+    #
+    # Returns true if the signature matches the signed request, otherwise false
+    def valid_callback_signature?(**options)
```
Comment (jnankin) [inline lib/phaxio/client.rb]: These should probably be argments, not options in a hash

File: lib/phaxio/client.rb [hunk 2]
```diff
@@ -271,15 +275,101 @@ def get_account_status
...
+
+    def generate_files_string(files)
+      return if files.nil?
+      files_array = files_to_array(files)
+      sorted_files = sort_files(files_array)
+      files_strings = sorted_files.map { |file| generate_file_string(file) }
+      files_strings.join
+    end
+
+    def strip_params(params)
```
Comment (jnankin) [inline lib/phaxio/client.rb]: Why does this have to be done?

File: lib/phaxio/client.rb (hunk 1 above)
Comment (jnankin) [inline lib/phaxio/client.rb]: Why do we need both valid and validate?

Comment (jtnegrotto) [inline lib/phaxio/client.rb]: Just for the sake of completion. You're right though, valid is all that's necessary.

File: README.md [hunk 3]
```diff
@@ -68,6 +68,35 @@ end: Time.now)`
...
+    class PhaxioCallbackExample < Sinatra::Base
+      Phaxio.config do |config|
+        config.api_key = '0123456789'
+        config.api_secret = '0123456789'
+        config.callback_token = '0123456789'
+      end
+
+      post '/phaxio_callback' do
+        if Phaxio.valid_callback_signature?(
+          request.env['HTTP_X_PHAXIO_SIGNATURE'],
```
Comment (jnankin) [inline README.md]: should just be X-Phaxio-Signature.  I assume this came from PHP, but this is the convention they used when dumping headers into $_SERVER

Comment (jnankin) [inline README.md]: nvm, looks like this is how sinatra does it too.

Comment (jtnegrotto) [inline README.md]: I wonder if it's specific to Sinatra or common to all Rack-based frameworks. Hopefully the latter.

**Pull request merged**